src/main/resources/assets/js/application/common/directives/tdprNewPasswordDirective.js

```diff
@@ -0,0 +1,48 @@
+define(['angular', 'application/common/tdprCommonModule'], function (angular, tdprCommonModule) {
+    tdprCommonModule.directive("newPassword", function () {
+        return {
+            restrict: 'AE',
+            templateUrl: '/html/partials/common/tdpr-directive-new-password.html',
+            scope: {
+                arePasswordsCorrect: '='
+            },
+            link: function (scope) {
+                scope.arePasswordsDifferent = false;
+                scope.isPasswordValid = true;
+                scope.newPassword = '';
+                scope.confirmPassword = '';
+
+                scope.$watch('[newPassword, confirmPassword]', function (newValue, oldValue, scope) {
+                    scope.arePasswordsDifferent = newValue[0] !== newValue[1];
+                    scope.isPasswordValid = isPasswordValid(newValue[0]);
+                    scope.arePasswordsCorrect = scope.isPasswordValid && !scope.arePasswordsDifferent;
+                });
+
+                function isPasswordValid(password) {
+                    var clues = [];
+                    if (password.length < 8) {
+                        clues.push("8 characters");
+                    }
+                    if (password.search(/[a-z]/) < 0) {
+                        clues.push("one lowercase letter.");
+                    }
+                    if (password.search(/[A-Z]/) < 0) {
+                        clues.push("one uppercase letter.");
+                    }
+                    if (password.search(/[0-9]/) < 0) {
+                        clues.push("one digit.");
+                    }
+                    if (password.search(/[!@#$%^&*]/) < 0) {
```

Comment:
I think `[^a-zA-Z0-9\s]` would suit this one better - it matches everything which is not a letter, digit or a whitespace character.